packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx

```diff
@@ -1041,7 +1042,14 @@ const TinyMCE = (props: NoteBodyEditorProps, ref: any) => {
 		async function onCopy(event: any) {
 			const copiedContent = editor.selection.getContent();
 			const copyableContent = getCopyableContent(copiedContent);
-			clipboard.writeHTML(copyableContent);
+
+			// In that case we need to set both HTML and Text context, otherwise it
+			// won't be possible to paste the text in, for example, a text editor.
+			// https://github.com/laurent22/joplin/issues/4788
+			clipboard.write({
+				text: htmlUtils.stripHtml(copiedContent),
+				html: copyableContent,
+			});
```

Comment:
When you copy and paste code from elsewhere like this, it makes it hard to maintain the code. Now whenever we want to improve or fix something related to copying text, there's two places to change and we have to remember that (and of course, we won't).

So please refactor the code so that the context menu and Ctrl+C run the same function. Create for example a utility function that takes HTML and, optionally, plain text as input, and copy that to the clipboard in the correct way.

Reply:
Got it, actually, I was also thinking of the same. Also, should I refactor to include Cut from context menu and (Ctrl X) in that function. Because technically in a cut we copy the text before erasing it.

Reply:
done, please review.